packages/sources/wrapped/src/endpoint/deposits.ts

```diff
@@ -0,0 +1,82 @@
+import { Requester, Validator, AdapterError } from '@chainlink/ea-bootstrap'
+import { Config, ExecuteWithConfig, InputParameters, AxiosResponse } from '@chainlink/types'
+
+export const supportedEndpoints = ['deposits']
+
+export interface ResponseSchema {
+  [token: string]: Address[]
+}
+
+export interface Networks {
+  [network: string]: string
+}
+
+export type Address = {
+  address: string
+  network: string
+  chainId: string
+}
+
+const customError = (data: any) => {
+  return typeof data !== 'object'
+}
+
+export const inputParameters: InputParameters = {
+  symbol: true,
+  network: false,
+  chainId: false,
+}
+
+const networks: Networks = {
+  BCH: 'bitcoincash',
+  BTC: 'bitcoin',
+  CELO: 'celo',
+  ETH: 'ethereum',
+  FIL: 'filecoin',
+  LTC: 'litecoin',
+  WCELO: 'ethereum', // Wrapped Celo
+  WCUSD: 'ethereum', // Wrapped Celo Dollar
+  WFIL: 'ethereum', // Wrapped Filecoin
+  WZEC: 'ethereum', // Wrapped Zcash
+  XLM: 'stellar',
+  XRP: 'ripple',
+  XTZ: 'tezos',
+  ZEC: 'zcash',
+  ZRX: 'ethereum', // 0x
+  cUSD: 'celo',
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const symbol = validator.validated.data.symbol
+  const chainId = validator.validated.data.chainId || 'mainnet'
+  const network = validator.validated.data.network || networks[symbol]
+  const url = `/deposits`
+
+  const options = { ...config.api, url }
+  const response = await Requester.request<ResponseSchema>(options, customError)
+  const addresses = response.data[symbol]
+
+  if (!addresses) {
+    const keys = Object.keys(response.data).join()
+    throw new AdapterError({
```

Comment:
we should add an integration test for the case when the request contains an invalid symbol.

Reply:
Since we nock the response from this endpoint http://chainlink.wrappedeng.com/deposits, which is a list of symbols with their respective addresses. What we do is only filter the symbol we need, I don't think that could be necessary to nock the error because the query is always made to the same endpoint without any extra parameter.